# Trinity, Mapping and Variant Calling 

## About

Took place on Scinet's [Niagara](https://docs.scinet.utoronto.ca/index.php/Niagara_Quickstart)
Also see ./Niagara .md

Software Tools Used: 
1. Samtools [samtools tutorial](http://quinlanlab.org/tutorials/samtools/samtools.html)
2. GATK/Picard
3. TrinityRNA seq 
4. Linux Tools
*** Usage examples in ./goodJobscripts

### Step 1: Trinity RNA seq

### Step 2: GATK best practises preprocessing for RNAseq Data 

[WORKFLOW](https://gatk.broadinstitute.org/hc/en-us/articles/360035531192-RNAseq-short-variant-discovery-SNPs-Indels-)




